api/sip/client.py

```diff
@@ -447,7 +448,7 @@ def end_session_message(
         timestamp = self.now()
 
         message = (code + timestamp +
-                   "AO" + institution_id + self.separator +
+                   "AO" + self.institution_id + self.separator +
```

Comment:
That's what's causing the `cannot concatenate 'str' and 'NoneType' objects` errors here. `None + "|"` is giving you an error.

Take a look at login_message:

```
        if location_code:
            message = message + self.separator + "CP" + location_code
```

If location_code is None then we don't even send that field. I think that's the right solution here. If you want to send this field even when the value is empty, then the default should be the empty string rather than None.

Reply:
I did see that this was likely the initial cause of the error, and I see that change takes care of my questions in Slack about needing to assign a blank/specific value in the Mock sip object. (My conditional assignment of a blank in `authenticator.py L1273` also takes care of that issue when values are captured from the DB patron auth/library configuration settings.)